### PR TITLE
Remove URLStreamHandlerRuntime

### PR DESCRIPTION
**Pull request by Godin**
I propose to remove `org.jacoco.core.runtime.URLStreamHandlerRuntime` for the following reasons:
* won't work in JDK 9 and actually already doesn't work with EA b148 due to use of `setAccessible` for private field `java.net.URL#handlers`
* has lower performance than `org.jacoco.core.runtime.ModifiedSystemClassRuntime` that is used by default and than `org.jacoco.core.runtime.SystemPropertiesRuntime`
* not used by default

### CODE REVIEW

File: org.jacoco.doc/docroot/doc/implementation.html [hunk 1]
```diff
@@ -205,7 +205,9 @@
     classes open a connection on this protocol. The returned connection object
     is the one that provides access to the coverage runtime through its
     <code>equals()</code> method. However to register the protocol the runtime
-    needs to access internal members of the <code>java.net.URL</code> class.</li> 
+    needs to access internal members of the <code>java.net.URL</code> class
+    and thus will require additional modifications to work with JDK 9
+    (at least was required for EA b148), so has been removed in 0.7.9.</li>
```
Comment (marchof) [inline org.jacoco.doc/docroot/doc/implementation.html]: Next JaCoCo release ist 0.8.0

**Comment by marchof** [comment]
@Godin What about merging this old PR? In the meantime we get this build warning for >= Java 9:

    WARNING: An illegal reflective access operation has occurred
    WARNING: Illegal reflective access by org.jacoco.core.runtime.URLStreamHandlerRuntime (file:/workspace/org.jacoco.core/target/org.jacoco.core-0.8.6-SNAPSHOT.jar) to field java.net.URL.handlers
    WARNING: Please consider reporting this to the maintainers of org.jacoco.core.runtime.URLStreamHandlerRuntime
    WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
    WARNING: All illegal access operations will be denied in a future release

**Comment by marchof** [comment]
@Godin I rebased this old PR on master. Can you please review?

**Comment by Godin** [comment]
@marchof thanks for rebasing and merging this! 👍

**Pull request merged**